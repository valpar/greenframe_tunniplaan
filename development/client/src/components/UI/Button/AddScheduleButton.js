import { faAngleRight } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const AddScheduleButton = (props) => {
  return (
    <div
      onClick={props.addScheduleHandler}
      className="realtive flex justify-between items-center group relative mx-auto mt-2 lg:mt-0 mb-3 w-1/3 h-11 font-bold bg-darkGray text-white shadow lg:w-full"
    >
      <div className="green-peeper" />
      <button type="button" className="w-full">
        LISA
      </button>
      <FontAwesomeIcon
        icon={faAngleRight}
        data-testid="font-awesome-icon"
        className={`${
          props.addSchedule ? "rotate-180" : ""
        } right-0 absolute text-base text-white p-4 duration-150`}
      />
    </div>
  );
};
