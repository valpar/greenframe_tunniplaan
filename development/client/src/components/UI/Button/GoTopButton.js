import { useEffect, useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCircleUp } from "@fortawesome/free-regular-svg-icons";

const GoTopButton = () => {
  const [showGoTop, setShowGoTop] = useState(false);

  const handleVisibleButton = () => {
    setShowGoTop(window.pageYOffset > 50);
  };

  const handleScrollUp = () => {
    window.scrollTo({ left: 0, top: 0, behavior: "smooth" });
  };

  useEffect(() => {
    window.addEventListener("scroll", handleVisibleButton);

    return () => window.removeEventListener("scroll", handleVisibleButton);
  }, []);

  return (
    <div
      className={
        showGoTop
          ? `sticky flex justify-center lg:justify-end bottom-4 md:bottom-1 z-10`
          : `hidden`
      }
      onClick={handleScrollUp}
    >
      <FontAwesomeIcon
        data-testid="go-top-button"
        className="block text-4xl md:text-6xl cursor-pointer text-darkGray opacity-70 hover:opacity-100 duration-200"
        icon={faCircleUp}
      />
    </div>
  );
};
// top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2
export default GoTopButton;
