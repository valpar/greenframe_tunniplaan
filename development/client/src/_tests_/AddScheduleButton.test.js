import React from 'react';
import { render, fireEvent, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { AddScheduleButton } from '../components/UI/Button/AddScheduleButton';
import { faAngleRight } from '@fortawesome/free-solid-svg-icons';

describe('AddScheduleButton', () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<AddScheduleButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the button with FontAwesomeIcon', () => {
    const addScheduleHandler = jest.fn();
    const addSchedule = true; 

    render(
      <AddScheduleButton
        addSchedule={addSchedule}
        addScheduleHandler={addScheduleHandler}
      />
    );

    const button = screen.getByText('LISA');
    const fontAwesomeIcon = screen.getByTestId('font-awesome-icon');

    expect(button).toBeInTheDocument();
    expect(fontAwesomeIcon).toBeInTheDocument();
    expect(fontAwesomeIcon).toHaveAttribute('data-icon', faAngleRight.iconName);

    fireEvent.click(button);
    expect(addScheduleHandler).toHaveBeenCalled();
  });
  
});
