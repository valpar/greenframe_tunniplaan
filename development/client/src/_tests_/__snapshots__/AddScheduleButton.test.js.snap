// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddScheduleButton matches snapshot 1`] = `
<div
  className="realtive flex justify-between items-center group relative mx-auto mt-2 lg:mt-0 mb-3 w-1/3 h-11 font-bold bg-darkGray text-white shadow lg:w-full"
>
  <div
    className="green-peeper"
  />
  <button
    className="w-full"
    type="button"
  >
    LISA
  </button>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-angle-right  right-0 absolute text-base text-white p-4 duration-150"
    data-icon="angle-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
